src/integrationTest/java/uk/gov/hmcts/reform/sscs/jms/listener/HmcHearingsEventTopicListenerItTest.java

```diff
@@ -0,0 +1,227 @@
+@SpringBootTest
+@AutoConfigureMockMvc(addFilters = false)
+@ActiveProfiles("integration")
+@TestPropertySource(locations = "classpath:config/application_it.properties")
+public class HmcHearingsEventTopicListenerItTest {
+
+    private static final String CASE_ID = "1234123412341234";
+    private static final String PROCESSING_VENUE_1 = "Cardiff";
+
+    private HmcHearingsEventTopicListener hmcHearingsEventTopicListener;
+    @MockBean
+    private ObjectMapper mapper;
+
+    @Autowired
+    private ProcessHmcMessageService processHmcMessageService;
+
+    @Autowired
+    private ProcessHmcMessageServiceV2 processHmcMessageServiceV2;
+
+    @MockBean
+    private JmsBytesMessage bytesMessage;
+
+    @MockBean
+    private IdamService idamService;
+    @MockBean
+    private CcdCaseService ccdCaseService;
+
+    @MockBean
+    private VenueService venueService;
+
+    @MockBean
+    private HmcHearingApi hearingApi;
+
+    @MockBean
+    private UpdateCcdCaseService updateCcdCaseService;
+
+    @Test
+    public void testHearingsUpdate() throws JMSException, HmcEventProcessingException, JsonProcessingException, CaseException, MessageProcessingException {
+        hmcHearingsEventTopicListener = new HmcHearingsEventTopicListener("BBA3", processHmcMessageService, processHmcMessageServiceV2);
+        ReflectionTestUtils.setField(hmcHearingsEventTopicListener, "objectMapper", mapper);
+        ReflectionTestUtils.setField(hmcHearingsEventTopicListener, "processEventMessageV2Enabled", false);
+
+        HmcMessage hmcMessage = createHmcMessage("BBA3");
+        IdamTokens idamTokens = IdamTokens.builder().build();
+
+        SscsCaseDetails sscsCaseDetails = getSscsCaseDetails();
+        VenueDetails venueDetails = getVenueDetails();
+        HearingGetResponse hearingGetResponse = getHearingGetResponse();
+
+        when(bytesMessage.getStringProperty("hmctsDeploymentId")).thenReturn("test");
+        when(bytesMessage.getBodyLength()).thenReturn(20L);
+
+        when(mapper.readValue(any(String.class), eq(HmcMessage.class))).thenReturn(hmcMessage);
+        when(idamService.getIdamTokens()).thenReturn(idamTokens);
+        when(ccdCaseService.getCaseDetails(anyLong())).thenReturn(sscsCaseDetails);
+
+        when(hearingApi.getHearingRequest(any(), any(), any(), any(), any()))
+            .thenReturn(hearingGetResponse);
+        when(venueService.getVenueDetailsForActiveVenueByEpimsId(any())).thenReturn(venueDetails);
+
+        hmcHearingsEventTopicListener.onMessage(bytesMessage);
+
+        verify(idamService, times(2)).getIdamTokens();
+        verify(ccdCaseService).getCaseDetails(anyLong());
+
+        verify(venueService).getVenueDetailsForActiveVenueByEpimsId(any());
+
+        verify(hearingApi).getHearingRequest(any(), any(), any(), any(), any());
+
+        verify(ccdCaseService).updateCaseData(
+            eq(sscsCaseDetails.getData()),
+            any(),
+            any(),
+            any());
+
```

Comment:
Could we verify that UpdateCcdCaseService updateCaseV2DynamicEvent is not invoked?